File: app/src/main/java/hu/bencelaszlo/notes/MainActivity.java
package hu.bencelaszlo.notes;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

import hu.bencelaszlo.notes.R;
import hu.bencelaszlo.notes.Adapters.NoteAdapter;
import hu.bencelaszlo.notes.Model.NoteModel;
import hu.bencelaszlo.notes.Utils.DatabaseHandler;

import java.util.Collections;
import java.util.List;
import java.util.Objects;

public class MainActivity extends AppCompatActivity implements DialogCloseListener{
    private DatabaseHandler db;
    private RecyclerView recyclerView;
    private NoteAdapter noteAdapter;
    private FloatingActionButton fab;
    private List<NoteModel> noteList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Objects.requireNonNull(getSupportActionBar()).hide();

        db = new DatabaseHandler(this);
        db.openDatabase();

        recyclerView = findViewById(R.id.noteListRecyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        noteAdapter = new NoteAdapter(db,MainActivity.this);
        recyclerView.setAdapter(noteAdapter);

        ItemTouchHelper itemTouchHelper = new
                ItemTouchHelper(new RecyclerItemTouchHelper(noteAdapter));
        itemTouchHelper.attachToRecyclerView(recyclerView);

        fab = findViewById(R.id.fab);

        noteList = db.getNotes();
        Collections.reverse(noteList);

        noteAdapter.setNotes(noteList);

        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                AddNote.newInstance().show(getSupportFragmentManager(), AddNote.TAG);
            }
        });
    }

    @Override
    public void handleDialogClose(DialogInterface dialog){
        noteList = db.getNotes();
        Collections.reverse(noteList);
        noteAdapter.setNotes(noteList);
        noteAdapter.notifyDataSetChanged();
    }
}

File: app/src/main/java/hu/bencelaszlo/notes/MainActivity.kt
package hu.bencelaszlo.notes

import android.os.Bundle
import android.content.DialogInterface
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import hu.bencelaszlo.notes.adapters.NoteAdapter
import com.google.android.material.floatingactionbutton.FloatingActionButton
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.ItemTouchHelper
import hu.bencelaszlo.notes.model.NoteModel
import hu.bencelaszlo.notes.utils.DatabaseHandler
import java.util.*

class MainActivity : AppCompatActivity(), DialogCloseListener {
    private lateinit var db: DatabaseHandler
    private var recyclerView: RecyclerView? = null
    private var noteAdapter: NoteAdapter? = null
    private var fab: FloatingActionButton? = null
    private lateinit var noteList: MutableList<NoteModel>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Objects.requireNonNull(supportActionBar)?.hide()
        db = DatabaseHandler(this)
        db.openDatabase()
        recyclerView = findViewById(R.id.noteListRecyclerView)
        recyclerView?.layoutManager = LinearLayoutManager(this)
        noteAdapter = NoteAdapter(db!!, this@MainActivity)
        recyclerView?.adapter = noteAdapter
        val itemTouchHelper = ItemTouchHelper(
            RecyclerItemTouchHelper(
                noteAdapter!!
            )
        )
        itemTouchHelper.attachToRecyclerView(recyclerView)
        fab = findViewById(R.id.fab)
        noteList = db.notes
        noteList.reverse()
        noteAdapter!!.setNotes(noteList)
        fab?.setOnClickListener {
            AddNote.newInstance().show(
                supportFragmentManager, AddNote.TAG
            )
        }
    }

    override fun handleDialogClose(dialog: DialogInterface?) {
        noteList = db.notes
        noteList.reverse()
        noteAdapter!!.setNotes(noteList)
        noteAdapter!!.notifyDataSetChanged()
    }
}